common/buildcraft/builders/tile/TileFiller.java

```diff
@@ -118,170 +117,118 @@ public void onPlacedBy(EntityLivingBase placer, ItemStack stack) {
-                .stream()
-                .filter(AddonFillingPlanner.class::isInstance)
-                .findFirst()
-                .orElse(null);
-            if (addon != null) {
-                volumeBox.locks.add(
-                    new Lock(
-                        new Lock.Cause.CauseBlock(pos, blockState.getBlock()),
-                        new Lock.Target.TargetAddon(addon.getSlot()),
-                        new Lock.Target.TargetResize(),
-                        new Lock.Target.TargetUsedByMachine(
-                            Lock.Target.TargetUsedByMachine.EnumType.STRIPES_WRITE
-                        )
-                    )
-                );
-                volumeBoxes.markDirty();
-                addon.updateBuildingInfo();
-            } else {
-                box.reset();
-                box.setMin(volumeBox.box.min());
-                box.setMax(volumeBox.box.max());
-                volumeBox.locks.add(
-                    new Lock(
-                        new Lock.Cause.CauseBlock(pos, blockState.getBlock()),
-                        new Lock.Target.TargetResize(),
-                        new Lock.Target.TargetUsedByMachine(
-                            Lock.Target.TargetUsedByMachine.EnumType.STRIPES_WRITE
-                        )
-                    )
-                );
-                volumeBoxes.markDirty();
-                parameters.addAll(Filling.initParameters());
-                updateBuildingInfo();
+        EnumFacing facing = blockState.getValue(BlockBCBase_Neptune.PROP_FACING);
+        BlockPos volumePos = pos.offset(facing.getOpposite());
+        VolumeBox volumeBox = volumeBoxes.getBoxAt(volumePos);
+        if (volumeBox != null && volumeBox.box.isOnEdge(volumePos)) {
+            box.setMin(volumeBox.box.min());
+            box.setMax(volumeBox.box.max());
+            placedWithVolume = true;
+            sendNetworkUpdate(NET_BOX);
+        } else {
+            TileEntity areaProvider = getNeighbourTile(facing.getOpposite());
+            if (areaProvider != null) {
+                ITileAreaProvider provider = areaProvider.getCapability(TilesAPI.CAP_TILE_AREA_PROVIDER, facing);
+                if (provider != null && provider.isValidFromLocation(pos)) {
+                    box.setMin(provider.min());
+                    box.setMax(provider.max());
+                    provider.removeFromWorld();
+                    sendNetworkUpdate(NET_BOX);
+                }
             }
-        } else if (tile instanceof IAreaProvider) {
-            IAreaProvider provider = (IAreaProvider) tile;
-            box.reset();
-            box.setMin(provider.min());
-            box.setMax(provider.max());
-            markerBox = true;
-            provider.removeFromWorld();
-            parameters.addAll(Filling.initParameters());
-            updateBuildingInfo();
-        }
-        sendNetworkUpdate(NET_RENDER_DATA);
-    }
-
-    @Override
-    public void validate() {
```

Comment:
Why remove SnapshotBuilder#validate and SnapshotBuilder#invalidate calls?

Reply:
Woops, I've readded them. At some point.